api/controllers/v0.3/DependentAccountAPI.js

```diff
@@ -105,55 +154,123 @@ const DependentEligibilityAPI = (args) => {
     if (!ilsClient) {
       throw new NoILSClient(
         "ILS Client not set in the Dependent Eligibility API."
       );
     }
 
-    let response = {};
-    const patron = await getPatronFromILS(barcode);
-    // First, check that they have a valid ptype to be able to create
-    // dependent accounts.
-    const hasValidPtype = checkPType(patron.patronType);
+    if (!dependentBarcode) {
+      throw new Error(
+        "The dependent account has no barcode. Cannot update parent account."
+      );
+    }
 
-    if (hasValidPtype) {
-      // Great, they have a valid ptype. Now check that they have not reached
-      // the dependent account limit.
-      const canCreateDependents = checkDependentLimit(patron.varFields);
+    const varFields = parent.varFields || [];
+    let varField;
+    // parent
+    const xFieldTags = varFields.filter((obj) => obj.fieldTag === "x");
+    // No varFields were found, so we can assume the patron doesn't
+    // have any dependent accounts yet.
+    if (xFieldTags.length === 0) {
+      varField = { fieldTag: "x", content: `DEPENDENTS ${dependentBarcode}` };
+    } else {
+      // Check for a varField that has `content` in the form of:
+      // "DEPENDENTS x,x,x"
+      // where `x` is a barcode. First get the varField that has "DEPENDENTS".
+      const dependentsVarField = xFieldTags.find(
+        (obj) => obj.content.indexOf("DEPENDENTS") !== -1
+      );
 
-      if (!canCreateDependents) {
-        response["eligible"] = false;
-        response["description"] =
-          "This patron has reached the limit to create dependent accounts.";
+      // There are varFields with a fieldTag of "x" but none with "DEPENDENTS".
+      // We can assume the patron doesn't have any dependents already.
+      if (!dependentsVarField) {
+        varField = { fieldTag: "x", content: `DEPENDENTS ${dependentBarcode}` };
       } else {
-        response["eligible"] = true;
-        response["description"] = "This patron can create dependent accounts.";
+        // The value is already there. So now append the new barcode.
+        dependentsVarField.content += `,${dependentBarcode}`;
+        varField = dependentsVarField;
       }
-    } else {
-      response["eligible"] = false;
-      response["description"] = "This patron does not have an eligible ptype.";
+    }
+
+    // This field is hardcoded but we only expect to update a patron's account
+    // if they have a dependent to add.
+    const updatedFields = {
+      varFields: [varField],
+    };
+
+    const response = await ilsClient.updatePatron(parent.id, updatedFields);
+
+    if (response.status !== 204) {
+      // The record wasn't found and couldn't be updated.
+      throw new Error("The parent patron couldn't be updated.");
     }
 
     return response;
   };
 
+  /**
+   * formatDependentAddress(address)
+   * A dependent account has the address as its parent account. The address
```

Comment:
Do you mean "has the SAME address as its parent account"?

Reply:
Yep.